boefjes/app.py

```diff
@@ -68,44 +73,53 @@ def start_working(
     while True:
         try:
             queues = scheduler_client.get_queues()
-
-            # We do not target a specific queue since we start one runtime for all organisations
-            # and queue ids contain the organisation_id
-            queues = [q for q in queues if q.id.startswith(queue_to_handle.value)]
-
-            logger.debug(f"Found queues: {[queue.id for queue in queues]}")
-        except (RequestHTTPError, HTTPError, ConnectionError):
+        except HTTPError:
             # Scheduler is having issues, so make note of it and try again
             logger.exception("Getting the queues from the scheduler failed")
             time.sleep(10 * settings.poll_interval)  # But not immediately
-
             continue
 
+        # We do not target a specific queue since we start one runtime for all organisations
+        # and queue ids contain the organisation_id
+        queues = [q for q in queues if q.id.startswith(queue_to_handle.value)]
+
+        logger.debug(f"Found queues: {[queue.id for queue in queues]}")
+
         for queue in queues:
+            logger.info(f"Popping from queue {queue.id}")
+
             try:
-                logger.info(f"Popping from queue {queue.id}")
-                task = scheduler_client.pop_task(queue.id)
-            except (RequestHTTPError, HTTPError, ConnectionError):
-                logger.exception("Popping task from scheduler failed")
-                time.sleep(10 * settings.poll_interval)
-                continue
-            except Exception:
+                p_item = scheduler_client.pop_item(queue.id)
+            except HTTPError:
                 logger.exception("Popping task from scheduler failed")
                 time.sleep(10 * settings.poll_interval)
                 continue
 
-            if not task:
+            if not p_item:
                 logger.info(f"Queue {queue.id} empty")
                 continue
 
+            logger.info(f"Handling task[{p_item.data.id}]")
+
             try:
-                logger.info(f"Handling task[{task.data.id}]")
-                item_handler.handle(task.data)
-            except StopWorking:
-                logger.info("Stopping worker...")
-                return
+                item_handler.handle(p_item.data)
             except:  # noqa
-                logger.exception("An error occurred handling a scheduler item")
+                logger.exception("An error occurred handling scheduler item %s", p_item.data.id)
+
+                try:
+                    scheduler_client.patch_task(str(p_item.id), TaskStatus.FAILED)
+                except HTTPError:
+                    logger.exception("Could not patch scheduler task to failed")
+
+                if exit_on_error:
```

Comment:
Please change `except` to `except BaseException as e` and change this to `if exit_on_error or not isinstance(e, Exception)` so we always reraise the exceptions that aren't a subclass of Exception and must not be silenced.